extensions/ql-vscode/test/pure-tests/log-summary-parser.test.ts

```diff
@@ -0,0 +1,129 @@
+import { expect } from 'chai';
+import 'mocha';
+
+import { parseVisualizerData } from '../../src/pure/log-summary-parser';
+
+describe('Parsing Valid Summary Text', () => {
+    const validSummaryText = `
```

Comment:
At the risk of complicating your test code by introducing file reading, I think it would be useful to keep the JSON test data in standalone files checked into the repo. Then each test case only needs to know the input file name and assertions on the expected output.

Reply:
I've made this change. I think it would still be good to make the tests more robust (right now I am testing for the edge cases that I have paid attention to and caught, but it may be good to test an entire full summary file to make sure nothing in the JSON file crashes). This may be a pretty large file to check in though, and I'm not entirely sure what I'd test for other than that it doesn't crash — given that it would be kind of laborious to manually check how many objects I'd expect to be parsed out of it. 

Maybe it would be good to just continuously add individual test cases that test different configurations of the JSON objects? 🤔